### PR TITLE
Play list with proper auth

### PR DESCRIPTION
**Pull request by Dev-AkshatGupta**
Authorization and PrivateRouting with every P0 function is done Files needed to be reviewed
\src\contextAndReducers\AuthProvider.jsx
src\components\LogInForm
src\components\NavBar
src\components\Toast
src\components\SignUpForm
\src\pages\SignUpPage
src\pages\Log-In-Page
Thanks for reviewing

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -15,25 +15,24 @@ import {PlayListPage} from "./pages/PlayList/PlayListPage";
...
-  
-  },[])
+const {authState,authDispatch }=useAuthorization();
+useEffect(()=>{
+ 
+const timeoutID= setTimeout(()=> {authDispatch({
+        type: "TOAST",
+        payload: { display:"none", message: "", type: "" },
+   
+      });clearTimeout(timeoutID)},2000)
```
Comment (b0llu) [inline src/App.js]: suggestion - this can be written a little bit cleaner

File: src/App.js [hunk 2]
```diff
@@ -42,14 +41,16 @@ const {modalDisplay, setModalDisplay}=useDataValues();
...
         <Route path="/" element={<ExplorePage/>}/>
         <Route path="/singleVideo-page/:source" element={<SingleVideoPage/>}/>
-        <Route path="/watchLater-page" element ={<WatchLater/>}/>
-        <Route path="/history-page" element ={< History/>}/>
+        <Route path="/watchLater-page" element ={<PrivateRoute><WatchLater/></PrivateRoute>}/>
+        <Route path="/history-page" element ={ <PrivateRoute>< History/></PrivateRoute>}/>
         <Route path="/mock" element={<Mockman/>}/>
-        <Route path="/playlist-page/:id" element={<PlayListPage/>}/>
+       
+          <Route path="/playlist-page/:id" element={ <PrivateRoute><PlayListPage/></PrivateRoute>}/>
```
Comment (b0llu) [inline src/App.js]: Great work in using private routing

File: src/components/NavBar/DropDownBox.jsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            <NavLink
+              to="/logIn-page"
+              className={({ isActive }) =>
+                isActive
+                  ? "nav-drawer-items-padding active-aside-link"
+                  : "nav-drawer-items-padding"
+              }
+            >
+              Log-In
+            </NavLink>
```
Comment (b0llu) [inline src/components/NavBar/DropDownBox.jsx:40]: NavLink is a great addition to the project

**Review by b0llu — APPROVED** [review]
Suggested few changes, great work on this PR akshat. You can merge this

File: src/App.js (hunk 1 above)
Comment (Dev-AkshatGupta) [reply src/App.js]: Thanks for reviewing I will look into it

File: src/components/NavBar/DropDownBox.jsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+          </div>
+        </>
+      ) : (
+        <a
+          className="btn btn-outline-error"
+          onClick={() => {
```
Comment (mikesch803) [inline src/components/NavBar/DropDownBox.jsx:46]: guestlogin and logout is not working

**Review by mikesch803 — APPROVED** [review]
just look into these suggestions else everything is good

**Pull request merged**